### DISCUSSION TITLE
Non used indicator changes results.

### DISCUSSION THREAD

**Discussion by Nireus79** [kernc/backtesting.py#893] [General]
def init(self):
        close = self.data.Close
        self.L_atr = self.I(atr_wrapper, self.data.df)
        self.M_K = resample_apply('1H', k_wrapper, self.data.df)
        self.M_D = self.I(sma_indicator, pd.Series(self.M_K), 3, fillna=False)
        self.H_ema1 = resample_apply('6H', ema_indicator, close, self.ema1, fillna=False)
        self.H_ema2 = resample_apply('6H', ema_indicator, close, self.ema2, fillna=False)

    def next(self):
        close = self.data.Close
        high = self.data.High
        low = self.data.Low
        self.trend1W = close > self.H_ema1
        self.buy_flag1D = self.M_K > self.M_D > self.lower_bound
        self.sell_flag1D = self.M_K < self.M_D < self.upper_bound

I set my indicators for different timeframes under init. Then I do not use the second ema. 
The problem is that when I delete or comment the second ema from init section I get different results. 
It is an indicator I do not call anywhere within my code and still changes results only because it is calculated. 
ema2 is also giving different results when optimized. 
What do I do wrong? I cannot figure it out. Please guide me.
Thank you.

**Comment by kernc** [comment]
See https://github.com/kernc/backtesting.py/issues/674 or https://github.com/kernc/backtesting.py/issues/732. Supposedly fixed in https://github.com/kernc/backtesting.py/commit/d7eaa459f8a0271552466b9eedb449c081ce63bd, to be released.

**Comment by Nireus79** [comment]
Thank you very much.
